http.go

```diff
@@ -17,61 +17,60 @@ const API = "api"
 
 /* The methods in this file provide assumption-ridden HTTP calls for Starr apps. */
 
+// Request contains the GET and/or POST values for an HTTP request.
+type Request struct {
+	URI   string     // path portion of the URL.
+	Query url.Values // GET parameters work for any request type.
+	Body  io.Reader  // Used in PUT, POST, DELETE. Not for GET.
+	uri   string     // derived URI.
```

Comment:
just a thought, is `uri` needed? I suppose you did this because you don't want to override the `URI` value, since we're working with pointers, right? but its scope is so small, it makes me think if it should be here. Thoughts?

Reply:
I ho-hum'd on this, and you nailed exactly my thought process. I didn't want to overwrite the input because it's a pointer. I originally made the interface input pointers because I wanted to be able to pass `nil`, but now that `uri` is in this struct it doesn't need to be a pointer because it can never be nil. Let me go ahead and re-do that and push it. I think it'll clear some of this up a bit.